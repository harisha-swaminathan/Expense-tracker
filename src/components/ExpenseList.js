import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div className="container"> 
  <div className="list-header">
    <div className="visible-mobile">Expenses</div>
    <div className="visible-desktop">Expense</div>
    <div className="visible-desktop">Amount</div>
  </div>
  <div className="list-bottom">
    {
      props.expenses.length === 0 ? (
        <div> 
          <span className=" list-item list-item--empty">
           No expenses
          </span>
        </div>
      ) : (
          props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />;
          })
        )
    }
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
