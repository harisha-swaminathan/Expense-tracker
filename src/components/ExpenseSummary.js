import React from 'react';
import { connect } from 'react-redux';
import {Link} from 'react-router-dom';
import ExpenseTotal from '../selectors/Expenses-total';
import selectExpenses from '../selectors/expenses';


export class ExpenseSummary extends React.Component{
  render(){
    const expense_s =this.props.expenses.length === 1 ? 'expense' : 'expenses' ;
    return(
        <div className="page-header">
            <div className="container">
                <h1 className="page-header__title">
                    Viewing <span>{this.props.expenses.length}</span> {expense_s} totalling <span>{this.props.expensesTotal}</span>
                    <div className="page-header__actions">
                        <Link className="button-color" to="/create">Add Expense</Link>
                    </div>
                </h1>
            </div>
        </div>
    );
}
}

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
    expensesTotal: <ExpenseTotal/>
  };
};

 export default connect(mapStateToProps)(ExpenseSummary);

 