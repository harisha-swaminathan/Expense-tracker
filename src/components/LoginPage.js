import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage=(props)=>(
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expense Tracker</h1>
     <button className="button-color" onClick={props.startLogin}>Login with Google</button>
     <a href="#">Try the app without logging in</a>
    </div>
        
      </div>
);

const mapDispatchToProps =(dispatch)=>({
  startLogin: ()=>dispatch(startLogin())
});

export default connect(undefined,mapDispatchToProps)(LoginPage);