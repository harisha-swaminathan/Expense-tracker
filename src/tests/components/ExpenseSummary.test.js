import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should correctly render ExpenseSummary with 1 expense',()=>{
    const wrapper=shallow(<ExpenseSummary expenses={length=1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot();
});
test('should correctly render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenses={length=23} expensesTotal={23512340987} />);
    expect(wrapper).toMatchSnapshot();
  });