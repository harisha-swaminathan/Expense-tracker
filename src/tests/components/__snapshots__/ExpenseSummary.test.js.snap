// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        235
      </span>
      <div
        className="page-header__actions"
      >
        <Link
          className="button-color"
          replace={false}
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </h1>
  </div>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        23512340987
      </span>
      <div
        className="page-header__actions"
      >
        <Link
          className="button-color"
          replace={false}
          to="/create"
        >
          Add Expense
        </Link>
      </div>
    </h1>
  </div>
</div>
`;
